### DISCUSSION TITLE
Baram Mesh for multi-region mesh preparation

### DISCUSSION THREAD

**Discussion by carowjp** [nextfoam/baram#60] [General]
Hello,
We are looking for a workflow for CHT problems from CAD geometry to multi-region mesh for OF. The tutorial CHT case uses geometry created in Baram Mesh. If the geometry comes from STL files can Baram Mesh create the interfaces between regions? Or must this step be done explicitly in a CAD package then imported as flow domain, solid domain, and interface? 
Thank you.

**Comment by kdg5424** [comment]
Hi @carowjp. This is donggyu.

Thank you for your interest in BARAM.
Here is the answers on your ask.

You should make the interface between solid domain and fluid domain in CAD package.
But If you make just one interface, then baramMesh seperates it to the fluid domain and solid domain respectively and automatically.
You may understand more easily with the picture.

![image](https://github.com/nextfoam/baram/assets/133930861/315cea07-ef69-4efc-bd06-bfc6e10ea3a2)

**Reply by carowjp** [reply]
Thank you for your fast reply.

**Comment by jiban** [comment]
Dear @carowjp,
You don't need to do additional work in a CAD package.
You just need to construct geometry as usual.
Here is a simple geometry for multi-region *BaramMesh* project.

![image](https://github.com/nextfoam/baram/assets/1674177/3d7e101f-4f49-47d8-a366-af5d208a4d16)

To clarify what surfaces are necessary from a CAD package, here is an exploded view of the geometry.

![image](https://github.com/nextfoam/baram/assets/1674177/4c138c3e-a373-474d-b804-4f68aa4879f3)

This surface is configured as *Inter-Region Interface* in BaramMesh.

![image](https://github.com/nextfoam/baram/assets/1674177/304b854d-b681-4f66-8e64-0137b15f6be8)

Finally, you have to specify internal points for each region.

![image](https://github.com/nextfoam/baram/assets/1674177/de4a1448-2a03-47b6-a6b6-8b91d6724436)
![image](https://github.com/nextfoam/baram/assets/1674177/a342fa93-1325-4e1b-bc93-8463d4d1f7ea)

After adding boundary layers, you can see two boundaries that make up one regional interface.

![image](https://github.com/nextfoam/baram/assets/1674177/845043bd-7e1f-4fcd-a9bd-e74d187c90ea)



Here is the *STL* files used for geometry.
[Geometry.zip](https://github.com/nextfoam/baram/files/14571703/Geometry.zip)

And this is the BaramMesh project file for the above sample.
[simpleMultiRegion.zip](https://github.com/nextfoam/baram/files/14571805/simpleMultiRegion.zip)

**Reply by carowjp** [reply]
Thank you for providing this helpful case! I was able to complete it myself. 

In the last image, I see you used a y-plane cut, which I did. I struggle to reproduce the exact view you had of the mesh. I hid all except InternalMesh and the InterRegionWall boundaries and have difficulty orienting. More for me to learn.

![Screenshot from 2024-03-13 10-13-48](https://github.com/nextfoam/baram/assets/6192645/a7c7bb06-0d2c-4b81-b15b-f4fbbb9a8579)

As suggestions for your software, it would be helpful in the Display Control list to indicate which boundaries or geometry is currently hidden by making the Name grey, otherwise you have to right click and check each to see the status Show/Hide. Similar with the Display Mode, perhaps there could be a small graphic indicating the mode for each boundary or geometry.

**Reply by jiban** [reply]
Hi, @carowjp,
I'm happy to hear that you got to the gist of BaramMesh in such a short time.
And thank you for the suggestion.
Actually, you can see if a boundary/geometry is hidden or displayed by looking at the color box at the right end of each item.
Hidden items have no color box while displayed items have a color box with its own color.
And you can easily acknowledge the display mode of an item because the item is shown in the graphics pane.
By the way, you may set the *Opacity* of internalMesh to *Opaque*(100% Opacity) to see the surface cells clearly.

**Reply by jiban** [reply]
However, you may still need some indication for display mode in the list.
If so, could you let me know the needs in detail?

**Reply by carowjp** [reply]
Thank you for more tips. The last trick I needed was using 'No Cut' on the interface. I opened an issue on your GitHub for the suggestions.
